flink-test-utils-parent/flink-migration-test-utils/src/main/java/org/apache/flink/test/migration/SnapshotGeneratorExecutor.java

```diff
@@ -0,0 +1,176 @@
+                                .value();
+                Method parametersMethod =
+                        migrationTestClass.getMethod(parametersMethodName, FlinkVersion.class);
+                executeParameterizedGenerateMethods(
+                        migrationTestClass, method, parametersMethod, flinkVersion);
+            }
+        }
+    }
+
+    private void executeGenerateMethods(
+            Class<?> migrationTestClass, Method method, FlinkVersion version) throws Throwable {
+        method.setAccessible(true);
+        List<TestRule> classRules = getRuleFields(migrationTestClass, ClassRule.class, null);
+
+        executeWithRules(
+                classRules,
+                new Statement() {
+                    @Override
+                    public void evaluate() throws Throwable {
+                        Object migrationTest = createMigrationTest(migrationTestClass);
+                        List<TestRule> rules =
+                                getRuleFields(migrationTestClass, Rule.class, migrationTest);
+                        executeWithRules(
+                                rules,
+                                new Statement() {
+                                    @Override
+                                    public void evaluate() throws Throwable {
+                                        method.invoke(migrationTest, version);
+                                    }
+                                });
+                    }
+                });
+    }
+
+    private void executeParameterizedGenerateMethods(
+            Class<?> migrationTestClass,
+            Method method,
+            Method parametersMethod,
+            FlinkVersion version)
+            throws Throwable {
+        method.setAccessible(true);
+        parametersMethod.setAccessible(true);
+        List<TestRule> classRules = getRuleFields(migrationTestClass, ClassRule.class, null);
+
+        executeWithRules(
+                classRules,
+                new Statement() {
+                    @Override
+                    public void evaluate() throws Throwable {
+                        Object migrationTest = createMigrationTest(migrationTestClass);
+                        List<TestRule> rules =
+                                getRuleFields(migrationTestClass, Rule.class, migrationTest);
+                        Collection<?> arguments =
+                                (Collection<?>) parametersMethod.invoke(migrationTest, version);
+                        for (Object argument : arguments) {
+                            executeWithRules(
+                                    rules,
+                                    new Statement() {
+                                        @Override
+                                        public void evaluate() throws Throwable {
+                                            method.invoke(migrationTest, version, argument);
```

Comment:
I'm not sure whether we actually need the version as a parameter here. Almost all of the test data generation methods have the FlinkVersion as a "unused parameter". The only class where we utilize this parameter is `TypeSerializerUpgradeTestBase`. But there, we pass this version to `createTestSpecifications` as well where it is used as an input parameter to each of the `TestSpecification` instances that are then used to call the actual test data genetation method (in the marked line, it's the `argument` parameter). Therefore, even in `TypeSerializerUpgradeTestBase` the flink version is present through the `TestSpecification` parameter. tbh, I haven't checked all the `TypeSerializerUpgradeTestBase` subclasses. But it is true for the ~5 classes I checked. Could we get rid of this parameter?

Reply:
Yes, we may first remove it since users could embed the target version in the specification. I'll first remove it